design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+}
+
+type NegotiableQuote {
+    billing_address: BillingAddress! @doc(description: "Billing address of negotiable quote")
+    attachements: [AttachmentContent] @doc(description: "Negotiable quote Attachements")
+    comments: [NegotiableQuoteComment] @doc(description: "Negotiable quote comments")
+    totals: TotalInfo @doc(description: "Negotiable quote totals info.")
+    payment_info: NegotiableQuotePaymentInfo! @doc(description: "Negotiable quote payment info.")
+}
+
+input NegotiableQuoteFilterInput {
+    ids: FilterEqualTypeInput @doc(description: "Filter Customer Negotibale quotes with an negotiable quote ID or list of negotiable quote IDs")
+    name: FilterMatchTypeInput @doc(description: "Filter by display name of the negotiable quote")
+}
+
+type BillingAddress {
+    id: ID! @doc(description: "The ID of the billing address")
+    firstname: String @doc(description: "The first name of the person associated with the billing address")
+    lastname: String @doc(description: "The family name of the person associated with the billing address")
+    customer_id: String @doc(description: "The customer ID")
+    email: String @doc(description: "The customer email id")
+    company: String @doc(description: "The customer's company")
+    telephone: String @doc(description: "The telephone number")
+    street: [String] @doc(description: "An array of strings that define the street number and name")
+    city: String @doc(description: "The city or town")
+    region: String @doc(description: "The region name")
+    region_id: String @doc(description: "The region ID")
+    region_code: String @doc(description: "The region code")
+    postcode: String @doc(description: "The customer's ZIP or postal code")
+    country_id: String @doc(description: "Deprecated: use `country_code` instead.")
+    same_as_billing: Int @doc(description: "Billing address confirmation")
+    save_in_address_book: Int @doc(description: "Save billing address in address book")
+}
+
+type AttachmentContent @doc(description: "Negotiable quote attachment file") {
+    base64_encoded_data: String
+    type: String
+    name: String!
+}
+
+type NegotiableQuoteComment {
+    entity_id: ID!
+    parent_id: ID!
+    creator_type: Int
+    is_decline: Int
+    is_draft: Int
+    creator_id: ID!
+    comment: String
+    created_at: String
+    attachments: [NegotiableQuoteCommentAttachments]
+}
+
+type NegotiableQuoteCommentAttachments {
+    attachment_id: ID!
+    comment_id: ID
+    file_name: String
+    file_path: String
+    file_type: String
+}
+
+type NegotiableQuotePaymentInfo {
+    payment_methods: [PaymentMethodsOutput]! @doc(description: "All list of payment options and totals")
+    totals: TotalsOutput! @doc(description: "List of totals")
+}
+
+type PaymentMethodsOutput {
+    code: String @doc(description: "Payment method code")
+    title: String @doc(description: "Payment method title")
+}
+
+type TotalsOutput {
```

Comment:
It seems like we just expose DB fields here. Please follow the structure similar to the one in `CartPrices` type. And ideally introduce generic reusable type/interface.

Reply:
Refactored the code similar to CartPrices